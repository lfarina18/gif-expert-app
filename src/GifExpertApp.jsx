import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
    const [categories, setCategories] = useState([
        'One Punch',
    ]);

    const onAddCategory = (newCategory) => {

        if (categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories]);
        // setCategories(cat => [newCategory, ...cat ]) // Otra forma de hacerlo.
    };

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                // setCategories={setCategories} 
                onNewCategory={onAddCategory}

            />


            {
                categories.map((category) => (
                    <GifGrid key={category}
                        category={category}
                    />
                ))
            }

        </>
    );
};
