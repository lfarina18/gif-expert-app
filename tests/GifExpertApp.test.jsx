import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas sobre el <GifExpertApp />', () => {
  test('Debe de hacer match con el snapshot <GifExpertApp />', () => {
    const { container } = render(<GifExpertApp />);

    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrarse el Heading h1', () => {

    render(<GifExpertApp />);

    expect(screen.getByRole('heading', { name: 'GifExpertApp' })).toBeTruthy();
  });



});
