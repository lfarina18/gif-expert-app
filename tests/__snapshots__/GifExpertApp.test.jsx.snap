// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el <GifExpertApp /> Debe de hacer match con el snapshot <GifExpertApp /> 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
